hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieSnapshotExporter.java

```diff
@@ -0,0 +1,213 @@
+          List<Tuple2<String, String>> filePaths = new ArrayList<>();
+          dataFiles.forEach(hoodieDataFile -> filePaths.add(new Tuple2<>(partition, hoodieDataFile)));
+
+          // also need to copy over partition metadata
+          Path partitionMetaFile =
+              new Path(new Path(cfg.sourceBasePath, partition), HoodiePartitionMetadata.HOODIE_PARTITION_METAFILE);
+          if (fs1.exists(partitionMetaFile)) {
+            filePaths.add(new Tuple2<>(partition, partitionMetaFile.toString()));
+          }
+
+          return filePaths.iterator();
+        }).foreach(tuple -> {
+          String partition = tuple._1();
+          Path sourceFilePath = new Path(tuple._2());
+          Path toPartitionPath = new Path(cfg.targetOutputPath, partition);
+          FileSystem ifs = FSUtils.getFs(cfg.targetOutputPath, serConf.newCopy());
+
+          if (!ifs.exists(toPartitionPath)) {
+            ifs.mkdirs(toPartitionPath);
+          }
+          FileUtil.copy(ifs, sourceFilePath, ifs, new Path(toPartitionPath, sourceFilePath.getName()), false,
+              ifs.getConf());
+        });
+
+        // Also copy the .commit files
+        LOG.info(String.format("Copying .commit files which are no-late-than %s.", latestCommitTimestamp));
+        FileStatus[] commitFilesToCopy =
+            fs.listStatus(new Path(cfg.sourceBasePath + "/" + HoodieTableMetaClient.METAFOLDER_NAME), (commitFilePath) -> {
+              if (commitFilePath.getName().equals(HoodieTableConfig.HOODIE_PROPERTIES_FILE)) {
+                return true;
+              } else {
+                String commitTime = FSUtils.getCommitFromCommitFile(commitFilePath.getName());
+                return HoodieTimeline.compareTimestamps(commitTime, latestCommitTimestamp,
+                    HoodieTimeline.LESSER_OR_EQUAL);
+              }
+            });
+        for (FileStatus commitStatus : commitFilesToCopy) {
+          Path targetFilePath =
+              new Path(cfg.targetOutputPath + "/" + HoodieTableMetaClient.METAFOLDER_NAME + "/" + commitStatus.getPath().getName());
+          if (!fs.exists(targetFilePath.getParent())) {
+            fs.mkdirs(targetFilePath.getParent());
+          }
+          if (fs.exists(targetFilePath)) {
+            LOG.error(
+                String.format("The target output commit file (%s targetBasePath) already exists.", targetFilePath));
+          }
+          FileUtil.copy(fs, commitStatus.getPath(), fs, targetFilePath, false, fs.getConf());
+        }
+      }
+    } else {
+      LOG.info("The job has 0 partition to copy.");
+    }
+  }
+
+  public static void main(String[] args) throws IOException {
+    // Take input configs
+    final Config cfg = new Config();
+    new JCommander(cfg, null, args);
+
+    // Create a spark job to do the snapshot export
+    SparkSession spark = SparkSession.builder().appName("Hoodie-snapshot-exporter").master("local[2]")
```

Comment:
```suggestion
    SparkSession spark = SparkSession.builder().appName("Hoodie-snapshot-exporter")
```

Reply:
if remove master() method, spark test unable to work properly. 
`org.apache.spark.SparkException: A master URL must be set in your configuration`

Reply:
ok but wouldn't this fail when running on production where we need to pass different master url?

Reply:
It's my fault. Thank you for pointing it out